HPC/WRF.md

```diff
@@ -0,0 +1,278 @@
+## WRF
+
+The Weather Research and Forecasting (WRF) model is one of the most used numerical weather prediction (NWP) system. WRF is used extensively for research and real-time forecasting. Large amount of computation resources are required for each simulation, especially for high resolution simulations. Compute and MPI performance are very important for WRF performance.
+
+## WRF correctness fix
+We have observed small numeric differences between WRF output models (v4.2.2) generated on Arm64 and intel platforms. The issue is caused by rounding error at last position (1ulp) from transcendental functions and several compiler optimization options. After identifying the root cause, we implemented [a patch](https://github.com/wrf-model/WRF/pull/1773) that allows bit-by-bit matching results on Graviton and intel processors. This patch works for both v4.2.2 and v4.5 (latest version). 
+This work confirms that the numerical divergence between arm64 and x86_64 is caused by round-up error and unsafe optimizations (FMA, ...) from the compiler. The output models on the two platforms are nearly identical. This type of numerical divergence are likely to show up again when using a different math library, a different version of math library, compiler. 
```

Comment:
Can delete this statement: "The output models on the two platforms are nearly identical."